R/PlainTextParam.R

```diff
@@ -177,38 +182,36 @@ setMethod("storeResults",
 setMethod("loadResults",
           signature(object = "MsExperiment",
                     param = "PlainTextParam"),
-          function(object, param, spectraPath = character()){
+          function(object, param, spectraPath = character()) {
               fl <- file.path(param@path, "sample_data.txt")
-              if (file.exists(fl)){ # should i have a error if does not exist ?
-                  sd <- read.table(fl)
-                  rownames(sd) <- NULL #read.table force numbering of rownames
-              }
-              s <- loadResults(Spectra(), param)
+              if (!file.exists(fl))
+                  stop("No 'sample_data.txt' file found in the provided path.")
+              sd <- read.table(fl, sep = "\t")
+              rownames(sd) <- NULL #read.table force numbering of rownames
+              s <- loadResults(Spectra(), param, spectraPath = spectraPath)
               res <- MsExperiment(sampleData = sd, spectra = s)
               validObject(res)
               res
-              }
-)
+          })
 
 #' @rdname PlainTextParam
 setMethod("loadResults",
           signature(object = "XcmsExperiment",
                     param = "PlainTextParam"),
-          function(object, param, spectraPath){
-              res <- callNextMethod() #check if need to add spectraPath = spectraPath
```

Comment:
it is possible to use either `callNextMethod(object = object, spectraPath = spectraPath)`, or simply `callNextMethod()` in which case all parameters passed to the original call will be passed along to the next method.